### PR TITLE
Upgrade Storybook to v7

### PR DESCRIPTION
**Pull request by jasverix**

### CODE REVIEW

**Review by mvelzel — APPROVED** [review]
I removed the addition of an unnecessary migration plugin that was added in this PR. Other than that it looks good to me!

File: package.json [hunk 1]
```diff
@@ -62,13 +63,16 @@
...
-    "eslint-plugin-storybook": "^0.6.10",
+    "eslint-plugin-storybook": "^0.6.11",
     "eslint-plugin-vue": "^9.3.0",
     "jsdom": "^20.0.3",
     "npm-run-all": "^4.1.5",
     "postcss": "^8.4.21",
     "prettier": "^2.7.1",
     "prettier-plugin-tailwindcss": "^0.2.2",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (laurensgroeneveld) [inline package.json:74]: Just double checking, do we really need react as a dev dependency?

File: src/components/BccButton/BccButton.stories.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+import { setup } from "@storybook/vue3";
 
-app.component("ChevronRightIcon", ChevronRightIcon);
-app.component("ChevronLeftIcon", ChevronLeftIcon);
-app.component("SearchIcon", SearchIcon);
+setup((app) => {
+  app.component("ChevronRightIcon", ChevronRightIcon);
+  app.component("ChevronLeftIcon", ChevronLeftIcon);
+  app.component("SearchIcon", SearchIcon);
+});
```
Comment (laurensgroeneveld) [inline src/components/BccButton/BccButton.stories.ts:11]: It doesn't render these icons for me on the button, other components seem to work fine.

**Review by laurensgroeneveld — COMMENTED** [review]
Two small comments.

File: src/components/BccButton/BccButton.stories.ts (hunk 2 above)
Comment (laurensgroeneveld) [reply src/components/BccButton/BccButton.stories.ts:11]: Checking again and it does show it... So not reproducable, maybe a caching issue.

**Pull request merged**